pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -292,5 +290,79 @@ func (rc *reconciler) attachDesiredVolumes() {
 			glog.Errorf(volumeToAttach.GenerateErrorDetailed("attacherDetacher.AttachVolume failed to start", err).Error())
 		}
 	}
+}
+
+// reportMultiAttachError sends events and logs situation that a volume that
+// should be attached to a node is already attached to different node(s).
+func (rc *reconciler) reportMultiAttachError(volumeToAttach cache.VolumeToAttach, nodes []types.NodeName) {
+	// Filter out the current node from list of nodes where the volume is
+	// attached.
+	// Some methods need []string, some other needs []NodeName, collect both.
+	// In theory, these arrays should have always only one element - the
+	// controller does not allow more than one attachment. But use array just
+	// in case...
+	otherNodes := []types.NodeName{}
+	otherNodesStr := []string{}
+	for _, node := range nodes {
+		if node != volumeToAttach.NodeName {
+			otherNodes = append(otherNodes, node)
+			otherNodesStr = append(otherNodesStr, string(node))
+		}
+	}
+
+	// Get list of pods that use the volume on the other nodes.
+	pods := rc.desiredStateOfWorld.GetVolumePodsOnNodes(otherNodes, volumeToAttach.VolumeName)
```

Comment:
Should this information be retrieved from desired state of world or actual state of world? It is possible that - multiple pods are requesting same volume on more than 1 node and they are all not getting attached because some other pod is actually using the volume.

Reply:
Problem is that ASW does not keep list of pods for attached volumes. It's reconstructed from node.status when the controller is restarted and we don't put pods there.

Reply:
I updated the comment below to:
```
// We did not find any pods that requests the volume. The pod must have been deleted already.
```
Message sent to user is: "Volume is already exclusively attached to one node and can't be attached to another" and IMO there is not much we could do about it.